authsignal/client.py

```diff
@@ -0,0 +1,256 @@
+import decimal
+import authsignal
+import authsignal.version
+
+import json
+import requests
+_UNICODE_STRING = str
+
+API_BASE_URL = 'https://signal.authsignal.com'
+
+class DecimalEncoder(json.JSONEncoder):
+    def default(self, o):
+        if isinstance(o, decimal.Decimal):
+            return (str(o),)
+        return super(DecimalEncoder, self).default(o)
+
+class Client(object):
+
+    def __init__(
+            self,
+            api_key=None,
+            api_url=API_BASE_URL,
+            timeout=2.0,
+            version=authsignal.version.VERSION,
+            session=None):
+        """Initialize the client.
+        Args:
+            api_key: Your Authsignal Secret API key of your tenant
+            api_url: Base URL, including scheme and host, for sending events.
+                Defaults to 'https://signal.authsignal.com'.
+            timeout: Number of seconds to wait before failing request. Defaults
+                to 2 seconds.
+        """
+        _assert_non_empty_unicode(api_url, 'api_url')
+        _assert_non_empty_unicode(api_key, 'api_key')
+
+        if api_key is None:
+            api_key = authsignal.api_key
+
+        self.session = session or requests.Session()
+        self.api_key = api_key
+        self.url = api_url
+        self.timeout = timeout
+        self.version = version
+
+    
+    def track_action(self, user_id, action_code, payload={}, path=None):
```

Comment:
```suggestion
    def track_action(self, user_id, action_code, payload=None, path=None):
```

Mutable default args are a common tripping point in Python